src/editor/underlying.rs

```diff
@@ -281,11 +199,42 @@ impl ObjectImpl for EditorWindow {
         }));
 
         drawing_area.connect_button_press_event(
-            clone!(@strong self.image as image, @strong obj => @default-return {warn!("A");Inhibit(false)}, move |_this, button| {
+            clone!(@strong self.image as image, @strong obj => @default-return Inhibit(false), move |this, button| {
+                tracing::warn!("Got button-press on drawing_area");
+                let mut image = image.borrow_mut();
+                let image = image.as_mut().unwrap();
+                image.operation_stack.start_operation_at(button.position().into());
+                this.queue_draw();
+                Inhibit(false)
+            }),
+        );
+
+        drawing_area.connect_motion_notify_event(
+            clone!(@strong self.image as image => @default-return Inhibit(false), move |this, motion| {
+                let primary_button_is_held = motion.state().contains(ModifierType::BUTTON1_MASK);
+                if primary_button_is_held {
+                    let mut image = image.borrow_mut();
+                    let image = image.as_mut().unwrap();
+                    image.operation_stack.update_current_operation_end_coordinate(motion.position().into());
+                    this.queue_draw();
+                }
+                Inhibit(false)
+            }),
+        );
+
+        drawing_area.connect_button_release_event(
+            clone!(@strong self.image as image, @strong obj => @default-return {warn!("A");Inhibit(false)}, move |this, button| {
                 tracing::warn!("y??");
                 if button.button() == BUTTON_PRIMARY {
                     let mut image = image.borrow_mut();
                     let image = image.as_mut().unwrap();
+                    if image.operation_stack.current_tool() != Tool::RectangleCrop {
```

Comment:
It is not really clear to me why we are checking against `Tool::RectangleCrop`, perhaps add a comment here explaining

Reply:
Yeah fair enough

RectangleCrop is a pretty bad name tbh, as I plan to make it autoselect windows later on.

The reason I check against RectangleCrop is because in this tool, clicking (or mouse release) should save the image (to clipboard or smth else).

Any suggestions for a better name?

Reply:
Well in that case, perhaps `CropAndSave` would be more on the lines with the thing it does?

Reply:
Yeah that sounds better, perhaps I'll add some doc comments to clarify them better too.